tomni/bbox_operations/check_overlap_bbox/main.py

```diff
@@ -0,0 +1,30 @@
+from typing import Union
+
+
+def check_overlap_bbox(bb1: Union[tuple, list], bb2: Union[tuple, list]) -> bool:
+
+    """
+    Checks if two bounding boxes overlap with eachother.
+    The bounding boxes are assumed the be structured as follows:
+    (xmin, ymin, xmax, ymax).
+    Warning: Touching bounding boxes are not seen as overlapping
+
+    Args:
+        bb1 (Union[tuple, list]): boundingbox 1, (x1, y1, x2 ,y2)
+        bb2 (Union[tuple, list]): boundingbox 2, (x1, y1, x2 ,y2)
+
+    Raises:
+        TypeError: Length of bounding box is the wrong size
+
+    Returns:
+        bool: if the bounding boxes are touching
+    """
+    if len(bb1) != 4 or len(bb2) != 4:
```

Comment:
add assert statement

Reply:
do you mean that i should replace the error statement with an assert. Because the nice thing about these kind of statement is that I can add what exactly went wrong. with assert statements that is not really possible